drivers/sensor/lis2dw12/lis2dw12_trigger.c

```diff
@@ -0,0 +1,178 @@
+			 enable);
+
+	return lis2dw12->hw_tf->update_reg(lis2dw12,
+		 LIS2DW12_CTRL5_ADDR,
+		 LIS2DW12_INT2_DRDY,
+		 enable);
+}
+
+/**
+ * lis2dw12_trigger_set - link external trigger to event data ready
+ */
+int lis2dw12_trigger_set(struct device *dev,
+			  const struct sensor_trigger *trig,
+			  sensor_trigger_handler_t handler)
+{
+	struct lis2dw12_data *lis2dw12 = dev->driver_data;
+	u8_t raw[6];
+
+	if (trig->chan == SENSOR_CHAN_ACCEL_XYZ) {
+		lis2dw12->handler_drdy = handler;
+		if (handler) {
+			/* dummy read: re-trigger interrupt */
+			lis2dw12->hw_tf->read_data(lis2dw12,
+					    LIS2DW12_OUT_X_L_ADDR, raw,
+					    sizeof(raw));
+			return lis2dw12_enable_int(dev, LIS2DW12_EN_BIT);
+		} else {
+			return lis2dw12_enable_int(dev, LIS2DW12_DIS_BIT);
+		}
+	}
+
+	return -ENOTSUP;
+}
+
+/**
+ * lis2dw12_handle_interrupt - handle the drdy event
+ * read data and call handler if registered any
+ */
+static void lis2dw12_handle_interrupt(void *arg)
+{
+	struct device *dev = arg;
+	struct lis2dw12_data *lis2dw12 = dev->driver_data;
+	struct sensor_trigger drdy_trigger = {
+		.type = SENSOR_TRIG_DATA_READY,
+	};
+	const struct lis2dw12_device_config *cfg = dev->config->config_info;
+
+	if (lis2dw12->handler_drdy != NULL) {
+		lis2dw12->handler_drdy(dev, &drdy_trigger);
+	}
+
+	gpio_pin_enable_callback(lis2dw12->gpio, cfg->int_gpio_pin);
+}
+
+static void lis2dw12_gpio_callback(struct device *dev,
+				    struct gpio_callback *cb, u32_t pins)
+{
+	struct lis2dw12_data *lis2dw12 =
+		CONTAINER_OF(cb, struct lis2dw12_data, gpio_cb);
+	const struct lis2dw12_device_config *cfg = dev->config->config_info;
+
+	ARG_UNUSED(pins);
+
+	gpio_pin_disable_callback(dev, cfg->int_gpio_pin);
+
+#if defined(CONFIG_LIS2DW12_TRIGGER_OWN_THREAD)
+	k_sem_give(&lis2dw12->gpio_sem);
+#elif defined(CONFIG_LIS2DW12_TRIGGER_GLOBAL_THREAD)
+	k_work_submit(&lis2dw12->work);
+#endif /* CONFIG_LIS2DW12_TRIGGER_OWN_THREAD */
+}
+
+#ifdef CONFIG_LIS2DW12_TRIGGER_OWN_THREAD
+static void lis2dw12_thread(int dev_ptr, int unused)
+{
+	struct device *dev = INT_TO_POINTER(dev_ptr);
+	struct lis2dw12_data *lis2dw12 = dev->driver_data;
+
+	ARG_UNUSED(unused);
+
+	while (1) {
+		k_sem_take(&lis2dw12->gpio_sem, K_FOREVER);
+		lis2dw12_handle_interrupt(dev);
+	}
+}
+#endif /* CONFIG_LIS2DW12_TRIGGER_OWN_THREAD */
+
+#ifdef CONFIG_LIS2DW12_TRIGGER_GLOBAL_THREAD
+static void lis2dw12_work_cb(struct k_work *work)
+{
+	struct lis2dw12_data *lis2dw12 =
+		CONTAINER_OF(work, struct lis2dw12_data, work);
+
+	lis2dw12_handle_interrupt(lis2dw12->dev);
+}
+#endif /* CONFIG_LIS2DW12_TRIGGER_GLOBAL_THREAD */
+
+int lis2dw12_init_interrupt(struct device *dev)
+{
+	struct lis2dw12_data *lis2dw12 = dev->driver_data;
+	const struct lis2dw12_device_config *cfg = dev->config->config_info;
+	int ret;
+
+	/* setup data ready gpio interrupt (INT1 or INT2) */
+	lis2dw12->gpio = device_get_binding(cfg->int_gpio_port);
+	if (lis2dw12->gpio == NULL) {
+		SYS_LOG_DBG("Cannot get pointer to %s device",
```

Comment:
This should be LOG_DBG

Reply:
sure. All the LOG related changes will be applied in the PR update.